q2_demux/_summarize/_visualizer.py

```diff
@@ -40,16 +40,15 @@ def __init__(self, directory_format, paired):
         self.paired = paired
 
 
-def _link_sample_n_to_file(files, counts, subsample_ns):
+def _link_sample_n_to_file(id_reads, counts, subsample_ns):
     results = collections.defaultdict(list)
     for num in subsample_ns:
         total = 0
-        for file in files:
-            sample_name = os.path.basename(file).split('_', 1)[0]
-            total += counts[sample_name]
+        for read, sample_id in id_reads:
```

Comment:
For historical clarity: I think `sample_id` makes more sense here because it reflects the vocabulary used elsewhere, for example in the docs on the `fastqManifest` file types.  @ebolyen and I just discussed this and agreed that `sample_id` is fine so long as I make it consistent throughout the code in this file.